File: programmers/pg_lv2_숫자의표현.js
function solution(n) {
  let count = 0;
  let half = Math.ceil(n / 2);

  if (n % 2 != 0 && n != 1) count++;
  for (i = 1; i <= half; i += 2) {
    if (n % i == 0) count++;
  }
  return count;
}

File: python/programmers/pg_lv2_숫자의표현.py
def solution(n):
    if n == 1 or n == 2: return 1
    
    totalSum = 1
    start = 1
    end = 2
    count = 1
    
    while end <= n and start < end:
        if totalSum <= n:
            if totalSum == n: count += 1
            totalSum += end
            end += 1
        else:
            totalSum -= start
            start += 1
            
    return count